qiskit/synthesis/su4/siswap_decompose.py

```diff
@@ -0,0 +1,286 @@
+            # terms of SiSwap†, then invert the whole thing.
+            inverse_circuit = _can_circuit(x, y, -z)
+            inverse_circuit_with_siswap_dg = QuantumCircuit(2)
+            for instruction in inverse_circuit:
+                if isinstance(instruction.operation, SiSwapGate):
+                    inverse_circuit_with_siswap_dg.z(0)
+                    inverse_circuit_with_siswap_dg.append(SiSwapGate().inverse(), [0, 1])
+                    inverse_circuit_with_siswap_dg.z(0)
+                else:
+                    inverse_circuit_with_siswap_dg.append(instruction)
+
+            V = inverse_circuit_with_siswap_dg.inverse()
+            v_decomp = TwoQubitWeylDecomposition(Operator(V))
+            can = _remove_pre_post(V, v_decomp)
+            circuit = circuit.compose(can, [0, 1])
+        else:
+            # make a circuit using 1 SiSwap that is able to bring a red point to here
+            follow = QuantumCircuit(2)
+
+            # x -> x + pi/8
+            # y -> y + pi/8
+            follow = follow.compose(SiSwapGate(), [0, 1])
+
+            eigenphase_crossing = False
+            if x > np.pi / 8:  # green region
+                # y -> y + pi/8
+                # z -> z + pi/8
+                follow = follow.compose(YGate().power(1 / 2), [0], front=True)
+                follow = follow.compose(YGate().power(1 / 2), [1], front=True)
+                follow = follow.compose(YGate().power(-1 / 2), [0])
+                follow = follow.compose(YGate().power(-1 / 2), [1])
+                if y + z < np.pi / 4:  # eigenphase crossing condition: a_2 - pi/4 < a_3 + pi/4
+                    eigenphase_crossing = True
+                # corresponding red coordinates
+                y -= np.pi / 8
+                z -= np.pi / 8
+            else:  # purple region
+                # x -> x - pi/8
+                # z -> z + pi/8
+                follow = follow.compose(XGate().power(1 / 2), [0], front=True)
+                follow = follow.compose(XGate().power(1 / 2), [1], front=True)
+                follow = follow.compose(XGate().power(-1 / 2), [0])
+                follow = follow.compose(XGate().power(-1 / 2), [1])
+                follow = follow.compose(ZGate(), [0], front=True)
+                follow = follow.compose(ZGate(), [0])
+                if y + z < np.pi / 8:  # eigenphase crossing condition: a_2 - pi/4 < a_3
+                    eigenphase_crossing = True
+                # corresponding red coordinates
+                x += np.pi / 8
+                z -= np.pi / 8
+
+            if eigenphase_crossing:
+                y, z = -z, -y
+
+            # final 3xSiSwap circuit: red --> fix crossing --> green or purple
+            red_decomp = _can_circuit(x, y, z)
+            circuit = circuit.compose(red_decomp, [0, 1])
+            if eigenphase_crossing:
+                # y, z -> -z, -y
+                circuit = circuit.compose(XGate().power(1 / 2), [0], front=True)
```

Comment:
Is there a reason to not use the `SX(dg)Gate` for this? 🙂

Reply:
hmm no its the same